lib/screens/speaker/speaker_repository.dart

```diff
@@ -50,3 +54,33 @@ class FakeSpeakerRepository implements SpeakerRepository {
     }
   }
 }
+
+class APISpeakerRepository extends SpeakerRepository {
+  final String classTag = "APIgetSpeakerRepository";
+  @override
+  Future<List<Speaker>> getAllSpeakers() async {
+    final String tag = classTag + "getAllSpeakers()";
+    http.Response response;
+    try {
+      response = await sl.get<http.Client>().get(S.getSpeakerUrl);
+    } catch (e) {
+      Log.e(tag: tag, message: "Network Error" + e.toString());
+      throw NetworkException();
+    }
+
+    if (response.statusCode == 200) {
+      Log.i(tag: tag, message: "Request Successful");
+      var speakerResponse = jsonDecode(response.body);
+      List<Speaker> speakerList;
+      (speakerResponse["data"] as List).map((e) => speakerList.add(Speaker.fromJson(e)));
```

Comment:
What map does is that it converts one type of datatype to another. What you are attempting to achieve is to be done using forEach.
So, you can either do this:
```suggestion
      (speakerResponse["data"] as List).forEach((e) => speakerList.add(Speaker.fromJson(e)));
```
Or this

```suggestion
      speakerList = (speakerResponse["data"] as List).map((e) => Speaker.fromJson(e)).toList();
```
Read a little more about [forEach](https://api.dart.dev/stable/1.10.1/dart-core/List/forEach.html) and [map](https://api.dart.dev/stable/1.10.1/dart-core/List/map.html) here.